### ISSUE TITLE
[4.x]: Restrict the usage of emoji

### ISSUE THREAD

**Issue by Megafry** [craftcms/cms#12935]
### Description

The feature #1627 that introduced emojis support can result in unintended consequences. For instance, I have an XML feed me import that contains a field with this specific notation for relations, "`:100:41:62:`", the field gets automatically transformed to "💯41:62:". 

### Steps to reproduce
1. Add the following text to a text field: "`:100: :1234:`"
2. Save the field.
3. The text is now :100: :1234:

### Expected behavior
it should be possible to specify whether emojis are allowed globally or at the field level. 

### Actual behavior
Emojis are allowed in all text fields and this can not be disabled:
`:happy:` =>  😄 

### Craft CMS version

4.4.3

### PHP version

8.0

### Installed plugins and versions

Feed Me - 5.1.0

**Comment by brandonkelly** [comment]
Thanks for reporting this!

We discussed several possible fixes today, none of which were especially cut and dry. It’s not as simple as adding a field/general setting for disabling emoji shortcodes, for a couple reasons:

1. Shortcodes are currently necessary for MySQL installs (until Craft 5 when we switch to `utf8mb4` table charsets by default – see #11823), and we don’t want to make people have to choose between emoji and avoiding this bug.
2. There’s all the existing data to consider. If a setting disables shortcode conversion, suddenly a bunch of emoji would start to appear as shortcodes.

The solution we settled on ended up being a little complex, and is going to require updates to Neo and Super Table to be fully realized (spicywebau/craft-neo#731, verbb/super-table#515), so I’ve made the fix for Craft 4.5 rather than the next 4.4 release.

(We’ll also need to add a new “Escape shortcodes” setting for Plain Text and Table fields to Feed Me, which will need to be enabled for any feeds that shouldn’t parse shortcodes.)

In the meantime, you can work around this by creating a module (e.g. using [Craft Generator](https://github.com/craftcms/generator/)), bootstrapped for every request, which configures LitEmoji to ignore the `:100:` and `:1234:` shortcodes:

```php
\LitEmoji\LitEmoji::config('excludeShortcodes', [
    '100',
    '1234',
]);
```

**Comment by brandonkelly** [comment]
Craft 4.5.0 is out with the fix for this.
